### DISCUSSION TITLE
Climate entity switch temperatur sensor

### DISCUSSION THREAD

**Discussion by owangen** [Koenkk/zigbee2mqtt#14282] [Devices]
I have implemented support the Namron touch thermostat for underfloor heating. This  thermostat supports 2 sensors, one on the device (reported as local_temp) and one from external sensor (typically placed in the floor, as external_temp). When i use the local_temp I want that to be shown in the climate entity, and when I use external sensor, I want external temperatur to be shown in the climate entity. In current implementation the local_temp is shown regardless of which temp sensor that is used. Is it possible to implement?

**Comment by chdefrene** [comment]
I have the same issue, albeit with a different device: https://github.com/Koenkk/zigbee2mqtt/discussions/14431

Ended up creating a custom adapter for setting your "external_temperature" directly on the climate entity. So far I have not been able to change it dynamically based on the "sensor" reading.
